chorus_lib/src/core.rs

```diff
@@ -149,7 +160,7 @@ pub trait Choreography<R = ()> {
     /// The method takes an implementation of `ChoreoOp`. Inside the method, you can use the operators provided by `ChoreoOp` to define a choreography.
     ///
     /// The method returns a value of type `R`, which is the return type of the choreography.
-    fn run(&self, op: &impl ChoreoOp) -> R;
+    fn run(self, op: &impl ChoreoOp) -> R;
```

Comment:
`run` must consume its ownership so the definition can move the ownership of fields.